29-Double Click Heart/script.js

```diff
@@ -0,0 +1,32 @@
+const loveMe = document.querySelector('.loveMe')
+const times = document.querySelector('#times')
+
+let timeClicked = 0
+
+loveMe.addEventListener('dblclick', (e) => {
+  createHeart(e)
+})
```

Comment:
if we are creating the arrow function just to encapsulate another function, then its perhaps best to not create the arrow function and to pass the original function directly 
ex.
`loveMe.addEventListener('dblclick', createHeart)` the `e` or `event` will be passed anyway.

Reply:
tried to do it and it gives me this error **( can't access lexical declaration 'createHeart' before initialization)**

Reply:
good thing, What do you think this error mean? make sure you dig into that.

there are many ways to fix this error here are two of them.
1. Move the function expression before this line of code.
 ex. 
 ```
 const createHeart = (e) => {...}
 loveMe.addEventListener('dblclick', createHeart)
 ```
2. Change the function from `function expression` to `function declaration`.
 ex. instead of `const createHeart = (e) => {...}` aka `expression`, change it to `function createHeart(e) {...}` aka `declaration`. with this approach the function doesn't need to be declared before line 6.

also might be very helpful for you to make some reading about the different between `function declaration`, `function expression` and `arrow function`, what are the pros and cons for each of them and when to use them.

Reply:
Thank you for reminding me about that